pkg/providers/ovirt/client/ovirt-client.go

```diff
@@ -88,7 +97,58 @@ func (client *richOvirtClient) GetVM(id *string, name *string, cluster *string)
 	return vm, nil
 }
 
-func (client *richOvirtClient) fetchVM(id *string, name *string, cluster *string) (*ovirtsdk.Vm, error) {
+// RenameVM rename the vm
+func (client *richOvirtClient) RenameVM(id string, newName string) error {
+	_, err := client.connection.SystemService().VmsService().VmService(id).
+		Update().
+		Vm(
+			ovirtsdk.NewVmBuilder().Name(newName).MustBuild(),
+		).
+		Send()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// StopVM stop the VM and wait for the vm to be stopped
+func (client *richOvirtClient) StopVM(id string) error {
+	vmService := client.connection.SystemService().VmsService().VmService(id)
+
+	// Return if VM is already stopped:
+	tmp, _ := vmService.Get().Send()
+	if status, _ := tmp.MustVm().Status(); status == ovirtsdk.VMSTATUS_DOWN {
```

Comment:
There is a chance that the VM with the given ID doesn't exist - `StopVM` might be called without any validation beforehand (i.e. `GetVM` call). `MustVm` can panic.

Reply:
Right, thanks. I will fix.